### DISCUSSION TITLE
Hope that deploy button can support framework & default env value setting.

### DISCUSSION THREAD

**Discussion by elkan1788** [vercel/community#1397] [Idea]
Thanks for Vercel guys offer the [deploy-button](https://vercel.com/docs/deploy-button) API that it is very useful & powerful.

Today I were try used this action that found something need to strong, such as:

- suggest that  enable project's framework setting
- suggest that it colud be set default value in env parameter
- make more powerful for [Snippets](https://vercel.com/docs/deploy-button#snippets) that some parameter were not supported

By the way,  I want to know why that can't work as set `project-name` and `repository-name` parameters.

Thanks.

**Comment by amyegan** [comment]
Thanks for the suggestions!

Environment variables cannot be set from the deploy button for security. You can include an [`envLink`](https://vercel.com/docs/deploy-button#environment-variables/environment-variables-link) to a document about how to get the correct values, or explain it in the [`envDescription`](https://vercel.com/docs/deploy-button#environment-variables/environment-variables-description).

The framework should be selected automatically for you when the project is created. But you can also override [Build Settings](https://vercel.com/docs/deploy-button#build-settings) for the deploy button as necessary. 

I'm not sure what `project-name` and `repository-name` parameters weren't working for you, but I'm happy to look for an answer if you can share what values you wanted to use.

**Reply by elkan1788** [reply]
> The framework should be selected automatically for you when the project is created. But you can also override [Build Settings](https://vercel.com/docs/deploy-button#build-settings) for the deploy button as necessary.

I know what your means, but actually I doesn't what this step to add build command, for example if it enable select framework parameter then can make sure the right environment to running my project. Because the automatica action what your said was not work well when I deploy the Hugo project  throght deploy button action.





> I'm not sure what `project-name` and `repository-name` parameters weren't working for you, but I'm happy to look for an answer if you can share what values you wanted to use.

There I mean following the document setted `project-name` and `repository-name` in deploy button's parameters, but when I click next by next step start deploy button, the actually these 2 parameters not working in this step, still need manual type in the project & repository name.

You can try this effect stage by this : [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fhugo-next%2Fhugo-theme-next-starter&env=HUGO_VERSION&envDescription=Enter%20latest%20version%20of%20Hugo%20engine.&envLink=https%3A%2F%2Fgithub.com%2Fgohugoio%2Fhugo%2Freleases%2F&project-name=my-blog&repo-name=my-blog&demo-title=Hugo%20NexT%20Theme&demo-description=Easily%20%26%20powerful%20theme%20of%20Hugo%20engine.&demo-url=https%3A%2F%2Fhugo-next.eu.org%2F&demo-image=https%3A%2F%2Fimgs.lisenhui.cn%2Fhugo-next%2Fhugo-next-demo.png)
